webuild-drafts/IntermediaryTesting.md

```diff
@@ -0,0 +1,538 @@
+  
+
+- Which attributes the RP is authorized to request
+  
+- The link between the RP and the intermediary
+  
+- Proof that the RP is using the intermediary’s services
+  
+
+    **Transaction Flow:**
+
+    - EUDIW receives a presentation request via the RPI
+
+    - EUDIW identifies the intermediary via its Access Certificate
+
+    - EUDIW verifies the Registration Certificate to confirm:
+
+        - The intermediary is authorized to request attributes on behalf of the RP
+
+        - The specific attributes being requested are authorized
+
+        - EUDIW displays to the user both the actual RP and the intermediary
+
+**Scenario 2: Member state does not issue Registration Certificates**
+
+According to the ARF, when registration certificates are not available:
+
+1. **Request Augmentation:** The intermediary includes in the presentation request:
+  
+
+- User-friendly name of the actual RP
+  
+- Unique identifier of the actual RP
+  
+- URL of the Registrar for the RP
+  
+- Identifier of the intended use
+  
+
+2. **Online Verification:** The EUDIW contacts the Registrar (at the provided URL) in real-time to verify:
+  
+
+- The RP’s registration status
+  
+- The relationship between the RP
+  and intermediary
+  
+- Authorized attribute access
+  
+
+**Critical Gap:** At the moment of writing, no standardized interface specification exists for EUDIW-to-Registrar online verification. This represents a significant interoperability challenge.
+
+**Cross-Border Complexity:**
+
+- RPs must be registered in their home member state
+  
+- RPIs with international operations must support registration mechanisms from multiple jurisdictions
+  
+- Each member state may implement different registration approaches on top of the choice whether to use Registration Certificates or not
+  
+
+## 5.2       Embedded Disclosure Policy Challenge
+
+**Challenge:** Electronic Attestations of Attributes (EAAs) may include embedded disclosure policies that specify which types of RPs are authorized to receive the attributes. The wallet must correctly evaluate these policies against the actual RP, not the intermediary.
+
+**Context:** Annex III of CIR 2024/2979 [i.9] specifies how embedded disclosure policies allow attribute issuers can limit what relying parties are allowed to receive attribute of the attestation:
+
+1. No policy
+  
+2. only authenticated relying parties which are explicitly listed in the disclosure
+  policy.
+  
+3. only authenticated wallet-relying parties with wallet-relying party access certificates derived from a specific root (or list of specific roots) or intermediate certificate(s).
+  
+
+The wallet verifies whether the wallet-relying party complies with the requirements of the embedded disclosure policy and informs the wallet user of the result.
+
+NOTE: The last option is based on the access certificate, not on the registration certificate. Since in the case of an intermediary, the access certificate identifies the RP
```

Comment:
there might be some sort of mix of filters upon access certificates and WRPRC minimizing the attributes to be allowed in the presentation